pkg/cmd/generate-bazel-extra/main.go

```diff
@@ -48,15 +49,54 @@ func runBuildozer(args []string) {
 	}
 }
 
-func getTestTargets(testTargetSize string) ([]string, error) {
-	if _, ok := testSizeToDefaultTimeout[testTargetSize]; !ok {
-		return nil, errors.New("testTargetSize should be one of {small,medium,large,enormous}")
+// parseQueryXML is used because Go doesn't support parsing XML "1.1".
+// It returns a map where the key is a test size {small,medium,large,enormous} and
+// the value is a list of test targets having that size.
+func parseQueryXML(data []byte) (map[string][]string, error) {
+	targetNameRegex, err := regexp.Compile(`<rule class="go_test" location=".*" name="(.*)">`)
```

Comment:
This is fine for now but it is a bit concerning to use regexp for parsing XML. One implication is that if we switch Bazel versions and the XML dump changes in format ever so slightly (for example, adding a space or changing the order of the attributes `class`, `location`, `name` -- changes that are totally legal and would produce an equivalent XML document) then this will stop working with a pretty confusing error message.

Reply:
I am thinking of two things we can do here, not that they will make us completely prone to the issues above but I think they will make us less prone to them.

1. We can change the regex to `<rule.*class="go_test".*name="(.*)".*>` to make it generalize better.

2. I am thinking of adding a test with a test case generated on demand from an empty package under testdata that contains an empty test target. We query `testdata/emptypackage` and ensure it runs without errors and returns the expected test target name and test target size.

Reply:
Also we can tag `bazci`'s test target with `no-cache` because I assume if a change breaks it, it won't be a change in the package or its dependencies but in bazel.

Reply:
> I am thinking of adding a test with a test case generated on demand from an empty package under testdata that contains an empty test target. We query testdata/emptypackage and ensure it runs without errors and returns the expected test target name and test target size.

Running `bazel query` from inside the test sandbox is not really going to be practical if that's what you're asking.

This runs in CI so presumably a Bazel upgrade would catch this as part of `Check generated code`. I think it's OK and not something we need to actively guard against right now.